### ISSUE TITLE
[release][ml-nightly] aws_ssh_sync and aws_durable_upload failing

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#20686]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

https://buildkite.com/ray-project/periodic-ci/builds/1758#7ce51ee1-4339-4f64-9b69-1e697da1dc18
https://buildkite.com/ray-project/periodic-ci/builds/1758#1ce6a022-ee60-4770-87b4-ca01b60c9926

```
After experiments: Invoking callback
--
  | Copied local node experiment dir /home/ray/ray_results/cloud_durable_upload to /tmp/tune_cloud_testan7oz8_0 for trial 27719_00001
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testpbo6vsaq/fn_trainable_27719_00002_2_score_multiplied=36_2021-11-23_10-28-30/checkpoint_000014 as 14 > 12
  | Copied bucket data from s3://data-test-ilr/durable_upload to /tmp/tune_cloud_testcpu_w378
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00000_0_score_multiplied=24_2021-11-23_10-28-29/checkpoint_000010 as 10 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00000_0_score_multiplied=24_2021-11-23_10-28-29/checkpoint_000012 as 12 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00001_1_score_multiplied=19_2021-11-23_10-28-30/checkpoint_000010 as 10 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00001_1_score_multiplied=19_2021-11-23_10-28-30/checkpoint_000012 as 12 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00002_2_score_multiplied=36_2021-11-23_10-28-30/checkpoint_000010 as 10 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00002_2_score_multiplied=36_2021-11-23_10-28-30/checkpoint_000012 as 12 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00003_3_score_multiplied=42_2021-11-23_10-28-30/checkpoint_000010 as 10 > 8
  | Skipping unobserved checkpoint: /tmp/tune_cloud_testcpu_w378/cloud_durable_upload/fn_trainable_27719_00003_3_score_multiplied=42_2021-11-23_10-28-30/checkpoint_000012 as 12 > 8
  | Traceback (most recent call last):
  | File "workloads/run_cloud_test.py", line 1130, in <module>
  | args.cpus_per_trial)
  | File "workloads/run_cloud_test.py", line 1118, in _run_test
  | test_durable_upload(bucket)
  | File "workloads/run_cloud_test.py", line 1060, in test_durable_upload
  | after_experiments_callback=after_experiments)
  | File "workloads/run_cloud_test.py", line 377, in run_resume_flow
  | after_experiments_callback()
  | File "workloads/run_cloud_test.py", line 1047, in after_experiments
  | bucket_dir_cp, for_driver_trial=2, for_worker_trial=2)
  | File "workloads/run_cloud_test.py", line 725, in assert_checkpoint_count
  | f"Trial {trial.trial_id} was not on the driver, "
  | AssertionError: Trial 27719_00000 was not on the driver, but did not observe the expected amount of checkpoints (0 != 2).
```

### Versions / Dependencies

Nightly test

### Reproduction script

n/a

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by ericl** [comment]
Is this a regression or an infrastructure issue (like https://github.com/ray-project/ray/issues/20616)?

**Comment by krfricke** [comment]
I'm rerunning it now as it might be general test flakiness. I'm not aware of any changes yesterday that could have caused that regression.
If this is just flakiness I'll try to improve this.

**Comment by ericl** [comment]
Ping on this? Is this still broken or just flaky?

**Comment by architkulkarni** [comment]
Looks like it's just flaky since it's passing recently on the master branch.

**Comment by krfricke** [comment]
Let's hope this is fixed by #23277
